concepts/functions/about.md

```diff
@@ -1,40 +1,167 @@
+## Return Values
+
+The function arguments are followed by zero or more return values which must also be explicitly typed.
+Single return values are left bare, multiple return values are wrapped in parenthesis.
+Values are returned to the calling code from functions using the [`return` keyword][return].
+There can be multiple `return` statements in a function.
+The execution of the function ends as soon as it hits one of those `return`s.
+If multiple values are to be returned from a function, they are comma seperated.
+More information about idiomatic use of [multiple return values][concept-multiple-return-values] can be found in the linked concept.
+
+```go
+func Hello(name string) string {
+  return "Hello " + name
+}
+
+func HelloAndGoodbye(name string) (string. string) {
+  return "Hello " + name, "Goodbye " + name
+}
+```
+
+## Invoking Functions
+
+Invoking a function is done by specifying the function name and passing arguments for each of the function's parameters in parenthesis.
+
+```go
+import "fmt"
+
+// No parameters, no return value
+func PrintHello() {
+    fmt.Println("Hello")
+}
+// Called like this:
+PrintHello()
+
+// One parameter, one return value
+func Hello(name string) string {
+  return "Hello " + name
+}
+// Called like this:
+greeting := Hello("Dave")
+
+// Multiple parameters, multiple return values
+func SumAndMultiply(a, b int) (int, int) {
+    return a+b, a*b
+}
+// Called like this:
+aplusb, atimesb := SumAndMultiply(a, b)
+```
+
+## Named Return Values and Naked Return
+
+As well as parameters, return values can optionally be named.
+If named return values are used, a `return` statement without arguments will return those values, this is known as a 'naked' return.
+
+```go
+func SumAndMultiplyThenMinus(a, b, c int) (sum, mult int) {
+    sum, mult = a+b, a*b
+    sum -= c
+    mult -= c
+    return
+}
+```
+
+## Pointers
+
+Functions in go pass their arguments by value.
+This means that the arguments are copied, and any changes to those arguments within the function will not be seen outside the function.
+
+```go
+val := 2
+func MultiplyByTwo(v int) int {
+    v = v * 2
+    return v
+}
+newval := MultiplyByTwo(val)
+// newval is 4, val is 2
+```
+
+To affect the values a function is called with, it is possible to pass pointer arguments, using the `*` syntax in the function, and the `&` syntax to generate a pointer.
+
+```go
+func swap(x, y *int) {
+    var temp int
+    temp = *x
+    *x = *y
+    *y = temp
+}
+
+var a int = 100
+var b int = 200
+swap(&a, &b)
+// a = 200, b = 100
+```
+
```

Comment:
There will be a separate exercise on pointers: https://github.com/exercism/go/issues/1643

```suggestion
```